inc/views/base/portal/portal.twig

```diff
@@ -0,0 +1,159 @@
+                        </tr>
+                        <tr>
+                            <td class="trow1">
+                                {% if mybb.user.uid %}
+                                    <span class="smalltext">
+                                        <em>{{ lang.member_welcome_lastvisit }}</em> {{ portal.lastvisit|raw }}<br />
+                                        {{ lang.since_then }}<br />
+                                        <strong>&raquo;</strong> {{ lang.new_announcements }}<br />
+                                        <strong>&raquo;</strong> {{ lang.new_threads }}<br />
+                                        <strong>&raquo;</strong> {{ lang.new_posts }}<br /><br />
+                                        <a href="{{ mybb.settings.bburl }}/search.php?action=getnew">{{ lang.view_new }}</a><br /><a href="{{ mybb.settings.bburl }}/search.php?action=getdaily">{{ lang.view_todays }}</a>
+                                    </span>
+                                {% else %}
+                                    <span class="smalltext">{{ lang.guest_welcome_registration|raw }}</span><br />
+                                    <br />
+                                    <form method="post" action="{{ mybb.settings.bburl }}/member.php">
+                                        <input type="hidden" name="action" value="do_login" />
+                                        <input type="hidden" name="url" value="{{ portal.portal_url }}" />
+                                        {{ portal.username }}<br />&nbsp;&nbsp;<input type="text" class="textbox" name="username" value="" /><br /><br />
+                                        {{ lang.password }}<br />&nbsp;&nbsp;<input type="password" class="textbox" name="password" value="" /><br /><br />
+                                        <label title="{{ lang.remember_me_desc }}"><input type="checkbox" class="checkbox" name="remember" value="yes" /> {{ lang.remember_me }}</label><br /><br />
+                                        <br /><input type="submit" class="button" name="loginsubmit" value="{{ lang.login }}" />
+                                    </form>
+                                {% endif %}
+                            </td>
+                        </tr>
+                    </table>
+                    <br />
+                {% endif %}
+                {% if portal.showpms %}
+                    <table border="0" cellspacing="{{ theme.borderwidth }}" cellpadding="{{ theme.tablespace }}" class="tborder">
+                        <tr>
+                            <td class="thead"><strong><a href="{{ mybb.settings.bburl }}/private.php">{{ lang.private_messages }}</a></strong></td>
+                        </tr>
+                        <tr>
+                            <td class="trow1">
+                                <span class="smalltext">{{ lang.pms_received_new|raw }}<br /><br />
```

Comment:
Rather than using `|raw` here, it looks like this is a fairly simple language variable that could be built in the template using the `trans()` function. One of my PRs from today marks the `trans()` function as being HTML safe, so as long as `$mybb->user['username']` is HTML safe, it should be fine here.